Godel.PK.BLL/Services/PortfolioService.cs

```
@@ -38,25 +38,31 @@ public async Task CreateAsync(RequestPortfolioViewModel portfolioViewModel, long
             await unitOfWork.SaveChangesAsync();
         }
 
-        public async Task<IEnumerable<ResponsePortfolioViewModel>> GetUserPortfoliosAsync(long userId)
+        public async Task<ResponsePortfoliosListViewModel> GetUserPortfoliosAsync(long userId)
         {
             var domainPortfolio = await repository.GetAllAsync(p => p.UserId == userId);
             List<ResponsePortfolioViewModel> result = new List<ResponsePortfolioViewModel>();
             var rates = await exchangeRateService.GetRatesAsync();
+            decimal totalAmount = 0;
             foreach (var portfolio in domainPortfolio)
             {
                 var rate = rates.First(r => r.Code == portfolio.CurrencyCode);
                 var mapped = portfolio.ToViewModel(rate.Value / rate.Scale);
                 mapped.CurrencyName = rate.Name;
                 result.Add(mapped);
+                totalAmount += mapped.LocalCurrencyAmount;
             }
                 
-            return result;
+            return new ResponsePortfoliosListViewModel
+            {
+                Portfolios = result,
+                TotalInLocalCurrency = totalAmount
+            };
         }
 
-        public async Task DeleteAsync(long portfolioId)
+        public async Task DeleteAsync(long portfolioId, long userId)
         {
-            var domainPortfolio = await repository.GetByIdAsync(portfolioId);
+            var domainPortfolio = await repository.GetSingleAsync(p => p.Id == portfolioId && p.UserId == userId);
             repository.Delete(domainPortfolio);
```

Why we changed this? Probably we should get userId from CurrentUserProvider and check whether we could remove and return unauthorized/forbiddedn? rather then fail with InvalidOperation on GetSingle()?